### PR TITLE
New Serviceline API modified, UPTSU API changes

### PR DESCRIPTION
**Pull request by indraniBan**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.  

# Type of change  

- [ ] Bug fix  
- [ ] New feature  
- [ ] Enhancement  
- [ ] Refactoring  
- [ ] Documentation  
- [ ] Other ( please specify )  

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/com/iemr/hwc/service/location/LocationServiceImpl.java [hunk 1]
```diff
@@ -343,17 +345,23 @@ private Map<String, Object> getDefaultLocDetails(ArrayList<Object[]> objList) {
...
-			}
+
+			for (Object[] objArr : objList) {
+
+				if (objArr[6] != null && objArr[7] != null) {
+					String[] villageIds = objArr[6].toString().split(",");
+					String[] villageNames = objArr[7].toString().split(",");
+					int villageIdSize = villageIds.length;
+					int villageNameSize = villageNames.length;
+					for (int i = 0; i < villageIdSize; i++) {
```
Comment (drtechie) [inline src/main/java/com/iemr/hwc/service/location/LocationServiceImpl.java:356]: Do you need a double loop here?

```
        if (villageIds.length == villageNames.length) {
            for (int i = 0; i < villageIds.length; i++) {
                Map<String, String> villageMap = new HashMap<>();
                villageMap.put("districtBranchID", villageIds[i]);
                villageMap.put("villageName", villageNames[i]);
                villageList.add(villageMap);
            }
        } else {
            // Handle the case where the number of IDs and names don't match. Or will that ever happen?
        }
```

File: src/main/java/com/iemr/hwc/service/ncdCare/NCDCareServiceImpl.java [hunk 2]
```diff
@@ -266,47 +267,124 @@ public Map<String, Long> saveBenVisitDetails(JsonObject visitDetailsOBJ, CommonU
...
-								.saveBenInvestigationDetails(wrapperBenInvestigationANC);
-
-					} else {
-						// Invalid Data..
+			int i = commonNurseServiceImpl.getMaxCurrentdate(benVisitDetailsOBJ.getBeneficiaryRegID(),
+					benVisitDetailsOBJ.getVisitReason(), benVisitDetailsOBJ.getVisitCategory());
+			if (i < 1) {
+				benVisitID = commonNurseServiceImpl.saveBeneficiaryVisitDetails(benVisitDetailsOBJ);
+
+				// 11-06-2018 visit code
```
Comment (drtechie) [inline src/main/java/com/iemr/hwc/service/ncdCare/NCDCareServiceImpl.java:275]: Unnecessary comment.

File: src/main/java/com/iemr/hwc/service/ncdCare/NCDCareServiceImpl.java [hunk 3]
```diff
@@ -266,47 +267,124 @@ public Map<String, Long> saveBenVisitDetails(JsonObject visitDetailsOBJ, CommonU
...
+								.fromJson(visitDetailsOBJ.get("investigation"), WrapperBenInvestigationANC.class);
+
+						if (wrapperBenInvestigationANC != null) {
+							wrapperBenInvestigationANC.setBenVisitID(benVisitID);
+							wrapperBenInvestigationANC.setVisitCode(benVisitCode);
+							investigationSuccessFlag = commonNurseServiceImpl
+									.saveBenInvestigationDetails(wrapperBenInvestigationANC);
+
+						} else {
+							// Invalid Data..
```
Comment (drtechie) [inline src/main/java/com/iemr/hwc/service/ncdCare/NCDCareServiceImpl.java:300]: The comment sounds like it is a grave problem, but there is no code that handles it.

File: src/main/java/com/iemr/hwc/service/ncdscreening/NCDScreeningServiceImpl.java [hunk 4]
```diff
@@ -457,33 +458,111 @@ public Map<String, Long> saveBenVisitDetails(JsonObject visitDetailsOBJ, CommonU
...
-					if (null != benChiefComplaintList && benChiefComplaintList.size() > 0) {
-						for (BenChiefComplaint benChiefComplaint : benChiefComplaintList) {
-							benChiefComplaint.setBenVisitID(benVisitID);
-							benChiefComplaint.setVisitCode(benVisitCode);
+			int i = commonNurseServiceImpl.getMaxCurrentdate(benVisitDetailsOBJ.getBeneficiaryRegID(),
+					benVisitDetailsOBJ.getVisitReason(), benVisitDetailsOBJ.getVisitCategory());
+			if (i < 1) {
+				benVisitID = commonNurseServiceImpl.saveBeneficiaryVisitDetails(benVisitDetailsOBJ);
+
+				// 11-06-2018 visit code
```
Comment (drtechie) [inline src/main/java/com/iemr/hwc/service/ncdscreening/NCDScreeningServiceImpl.java:466]: Unnecessary comment.

File: src/main/java/com/iemr/hwc/service/ncdscreening/NCDScreeningServiceImpl.java [hunk 5]
```diff
@@ -457,33 +458,111 @@ public Map<String, Long> saveBenVisitDetails(JsonObject visitDetailsOBJ, CommonU
...
-					commonNurseServiceImpl.saveBenChiefComplaints(benChiefComplaintList);
+
+					if (visitDetailsOBJ.has("cdss") && !visitDetailsOBJ.get("cdss").isJsonNull()) {
+						JsonObject cdssObj = visitDetailsOBJ.getAsJsonObject("cdss");
+						CDSS cdss = InputMapper.gson().fromJson(cdssObj, CDSS.class);
+						cdss.setBenVisitID(benVisitID);
+						cdss.setVisitCode(benVisitCode);
+
+						if (cdssObj.has("presentChiefComplaintDb")) {
+							JsonObject presentCheifComplaintObj = cdssObj.getAsJsonObject("presentChiefComplaintDb");
```
Comment (drtechie) [inline src/main/java/com/iemr/hwc/service/ncdscreening/NCDScreeningServiceImpl.java:494]: Same comment related DRYness as above.

File: src/main/java/com/iemr/hwc/service/neonatal/NeonatalServiceImpl.java [hunk 6]
```diff
@@ -435,6 +436,79 @@ private Map<String, Long> saveBenVisitDetails(JsonObject visitDetailsOBJ, Common
...
 					commonNurseServiceImpl.saveBenChiefComplaints(benChiefComplaintList);
 				}
+
+				if (visitDetailsOBJ.has("cdss") && !visitDetailsOBJ.get("cdss").isJsonNull()) {
+					JsonObject cdssObj = visitDetailsOBJ.getAsJsonObject("cdss");
+					CDSS cdss = InputMapper.gson().fromJson(cdssObj, CDSS.class);
+					cdss.setBenVisitID(benVisitID);
+					cdss.setVisitCode(benVisitCode);
+
+					if (cdssObj.has("presentChiefComplaintDb")) {
```
Comment (drtechie) [inline src/main/java/com/iemr/hwc/service/neonatal/NeonatalServiceImpl.java:446]: Same comment related DRYness as above.

File: src/main/java/com/iemr/hwc/service/pnc/PNCServiceImpl.java [hunk 7]
```diff
@@ -453,43 +455,120 @@ public Map<String, Long> saveBenVisitDetails(JsonObject visitDetailsOBJ, CommonU
...
 					BeneficiaryVisitDetail.class);
-			int i=commonNurseServiceImpl.getMaxCurrentdate(benVisitDetailsOBJ.getBeneficiaryRegID(),benVisitDetailsOBJ.getVisitReason(),benVisitDetailsOBJ.getVisitCategory());
-			if(i<1) {
-			benVisitID = commonNurseServiceImpl.saveBeneficiaryVisitDetails(benVisitDetailsOBJ);
+			int i = commonNurseServiceImpl.getMaxCurrentdate(benVisitDetailsOBJ.getBeneficiaryRegID(),
+					benVisitDetailsOBJ.getVisitReason(), benVisitDetailsOBJ.getVisitCategory());
+			if (i < 1) {
+				benVisitID = commonNurseServiceImpl.saveBeneficiaryVisitDetails(benVisitDetailsOBJ);
+
+				// 11-06-2018 visit code
```
Comment (drtechie) [inline src/main/java/com/iemr/hwc/service/pnc/PNCServiceImpl.java:463]: Unnecessary comment.

File: src/main/java/com/iemr/hwc/service/pnc/PNCServiceImpl.java [hunk 8]
```diff
@@ -453,43 +455,120 @@ public Map<String, Long> saveBenVisitDetails(JsonObject visitDetailsOBJ, CommonU
...
-			Long benVisitCode = commonNurseServiceImpl.generateVisitCode(benVisitID, nurseUtilityClass.getVanID(),
-					nurseUtilityClass.getSessionID());
+					// TODO Save Ben Upload Files
+					if (visitDetailsOBJ.has("cdss") && !visitDetailsOBJ.get("cdss").isJsonNull()) {
+						JsonObject cdssObj = visitDetailsOBJ.getAsJsonObject("cdss");
+						CDSS cdss = InputMapper.gson().fromJson(cdssObj, CDSS.class);
+						cdss.setBenVisitID(benVisitID);
+						cdss.setVisitCode(benVisitCode);
+
+						if (cdssObj.has("presentChiefComplaintDb")) {
```
Comment (drtechie) [inline src/main/java/com/iemr/hwc/service/pnc/PNCServiceImpl.java:499]: Same comment related DRYness as above.

File: src/main/java/com/iemr/hwc/service/pnc/PNCServiceImpl.java [hunk 9]
```diff
@@ -453,43 +455,120 @@ public Map<String, Long> saveBenVisitDetails(JsonObject visitDetailsOBJ, CommonU
...
+								BenAdherence.class);
+						benAdherence.setBenVisitID(benVisitID);
+						benAdherence.setVisitCode(benVisitCode);
+						commonNurseServiceImpl.saveBenAdherenceDetails(benAdherence);
+					}
 
-			// 11-06-2018 visit code
-			Long benVisitCode = commonNurseServiceImpl.generateVisitCode(benVisitID, nurseUtilityClass.getVanID(),
-					nurseUtilityClass.getSessionID());
+					// TODO Save Ben Upload Files
```
Comment (drtechie) [inline src/main/java/com/iemr/hwc/service/pnc/PNCServiceImpl.java:492]: is this TODO comment still valid?

**Review by ravishanigarapu — APPROVED** [review]
look ok

**Pull request merged**